src/Aspire.Hosting.Azure/Provisioning/Internal/PublishModeProvisioningContextProvider.cs

```diff
@@ -75,8 +79,40 @@ public override async Task<ProvisioningContext> CreateProvisioningContextAsync(J
         return await base.CreateProvisioningContextAsync(userSecrets, cancellationToken).ConfigureAwait(false);
     }
 
-    private async Task RetrieveAzureProvisioningOptions(CancellationToken cancellationToken = default)
+    private async Task RetrieveAzureProvisioningOptions(JsonObject userSecrets, CancellationToken cancellationToken = default)
     {
+        // Clear any options that might have been loaded from top-level configuration
+        // so we can read from values user secrets
+        _options.SubscriptionId = null;
+        _options.Location = null;
+        _options.ResourceGroup = null;
+        _options.AllowResourceGroupCreation = false;
```

Comment:
Mutating options like this feels bad, where else does this get read of injected? Can we instead change it before options creation?

Reply:
These get read in the base class and are used to construct the `ArmClient` and other provider-types that we use in the provisioner. To make this cleaner, we can update the base class to read the values from local state then only read from these options in the run-mode implementation and remove them from the publish-mode one.